src/models/state/mempool.rs

```diff
@@ -69,14 +87,20 @@ pub struct Mempool {
 
     /// Allows the mempool to report transactions sorted by [`FeeDensity`] in
     /// both descending and ascending order.
-    #[get_size(ignore)] // This is relatively small compared to `LookupTable`
+    #[get_size(ignore)] // This is relatively small compared to `tx_dictionary`
     queue: DoublePriorityQueue<Digest, FeeDensity>,
 
     /// Records the digest of the block that the transactions were synced to.
     /// Used to discover reorganizations.
     tip_digest: Digest,
 }
 
+/// note that all methods that modify state and result in a MempoolEvent
+/// notification are private or pub(super).  This enforces that these methods
+/// can only be called from/via GlobalState.
+///
+/// Mempool updates must go through GlobalState so that it can
+/// forward mempool events to the wallet in atomic fashion.
```

Comment:
right.  The point of MempoolEvent is to allow any type to be a mempool listener and to *avoid* linking WalletState and Mempool directly.

The comment is just to provide some app context for anyone reading mempool code and so they don't make mempool modifying methods pub in the future, which would allow bypassing GlobalState wrapper methods.